src/opm/io/eclipse/ESmry.cpp

```diff
@@ -219,40 +232,81 @@ ESmry::ESmry(const std::string &filename, bool loadBaseRunData)
         }
 
         std::string smspecFile = std::get<0>(smryArray[n]);
-        std::string unsmryFile = smspecFile.substr(0, smspecFile.size() - 6) + "UNSMRY";
 
-        EclFile unsmry(unsmryFile);
-        unsmry.loadData();
-
-        std::vector<EclFile::EclEntry> list1 = unsmry.getList();
-
-        // 2 or 3 arrays pr time step.
-        //   If timestep is a report step:  MINISTEP, PARAMS and SEQHDR
-        //   else : MINISTEP and PARAMS
+        rootN = formattedVect[n] ? smspecFile.substr(0, smspecFile.size() - 7) : smspecFile.substr(0, smspecFile.size() - 6);
```

Comment:
There must be some `basename` utility somewhere?

Reply:
it is possible to use boost::filesystem::path instead of std::string and do the following
```C++
boost::filesystem::path smspecFile(std::get<0>(smryArray[n]));
boost::filesystem::path rootN = smspecFile.parent_path() /  smspecFile.stem();
boost::filesystem::path unsmryFile = rootN += ".SMSPEC";
```
But there are several places/functions in ESmry  (which is already merged, not only this PR) where this should be implemented for consistency. 

I would like to include this in the upcoming release, and I would therefor like to do the full implementation of boost::filesystem::path later if we agree that this is what should be done. 

@bska do you have an opinion/recommendation on this issue ?

Reply:
class ESmry is now consistently using boost::filesystem::path instead of std::string. The constructor and a number of member functions have been updated in the last commit.